Ecommerce/ProductService/Repositories/InMemoryProductRepository.cs

```diff
@@ -0,0 +1,90 @@
+﻿using ProductService.Models;
+using ProductService.Validators;
+
+namespace ProductService.Repositories
+{
+    public class InMemoryProductRepository : IProductRepository
+    {
+        private static int IdCounter = 0;
+        private Dictionary<int, Product> _products = new Dictionary<int, Product>() { };
+        private ProductValidator _productValidator;
+
+        public InMemoryProductRepository(ProductValidator productValidator) 
+        {
+            _productValidator = productValidator;
+        }
+
+        public Dictionary<int, Product> GetProducts()
+        {
+            Dictionary<int, Product> sendingProducts = new Dictionary<int, Product>();
+
+            foreach (var product in _products)
+                sendingProducts.Add(product.Key, product.Value);
+
+            return sendingProducts;
+        }
+
+        public bool GetProduct(int id, out Product product)
+        {
+            product = null;
+            bool isFinded = false;
+
+            if (_products.ContainsKey(id))
+            {
+                isFinded = true;
+                product = _products[id];
+            }
```

Comment:
Можно использовать метод TryGetValue:

```suggestion
            if (_products.TryGetValue(id, out Product value))
            {
                isFinded = true;
                product = value;
            }
```